### PR TITLE
feat(VE): migrate to Vanilla Extract: box, stack, inline, touchable and text

### PR DESCRIPTION
**Pull request by sergivillar**

### CODE REVIEW

File: src/text.tsx [hunk 1]
```diff
@@ -128,28 +89,50 @@ export const Text: React.FC<TextProps> = ({
...
+        [styles.vars.mobileLineHeight]: mobileLineHeight ? pxToRem(mobileLineHeight) : '',
+        [styles.vars.desktopSize]: desktopSize ? pxToRem(desktopSize) : '',
+        [styles.vars.desktopLineHeight]: desktopLineHeight ? pxToRem(desktopLineHeight) : '',
+    });
+    const textVars = truncate
+        ? assignInlineVars({
+              [styles.vars.lineClamp]: String(lineClampValue),
+              [styles.vars.wordBreak]: wordBreak ? 'break-word' : 'inherit',
+          })
+        : assignInlineVars({[styles.vars.wordBreak]: wordBreak ? 'break-word' : 'inherit'});
```
Comment (atabel) [inline src/text.tsx:113]: maybe you are abusing css vars here? can't these be classes set conditionally

File: src/inline.tsx [hunk 2]
```diff
@@ -54,11 +38,21 @@ const Inline: React.FC<Props> = ({
...
+            className={classnames(className, {
+                [styles.fullWidth]: isFullWith,
+                [styles.noFullWidth]: !isFullWith,
+            })}
+            style={{
+                ...assignInlineVars({
+                    [styles.vars.space]: typeof space === 'number' ? `${space}px` : '',
+                }),
+                justifyContent: getJustifyContent(space),
+                alignItems,
```
[the remark this answers is not in the document]
Comment (sergivillar) [reply src/inline.tsx:54]: adding baseline to sprinkles

File: src/stack.tsx [hunk 3]
```diff
@@ -33,13 +17,13 @@ type Props = {
...
     const {space, className, children, role} = props;
-    const classes = useStyles({space});
 
     const isNumeric = typeof space === 'number';
 
     return (
         <div
-            className={classnames(className, isNumeric ? classes.marginStack : classes.flexStack)}
+            className={classnames(className, isNumeric ? styles.marginStack : styles.flexStack)}
+            style={assignInlineVars({[styles.vars.space]: isNumeric ? `${space}px` : `space-${space}`})}
```
[the remark this answers is not in the document]
Comment (sergivillar) [reply src/stack.tsx:26]: it should not, as marginStack class is only applied when isNumric, and marginTop style only applies when marginStack class is used

File: src/boxed.tsx [hunk 4]
```diff
@@ -54,16 +40,25 @@ export const Boxed = React.forwardRef<HTMLDivElement, Props>(
...
                 role={role}
                 {...getPrefixedDataAttributes(dataAttributes)}
                 aria-label={ariaLabel}
+                style={{
+                    borderRadius: 8,
+                    overflow: 'hidden',
+                    background:
+                        isInverseInside && !isDarkMode
+                            ? vars.colors.backgroundBrand
+                            : vars.colors.backgroundContainer,
```
Comment (atabel) [inline src/boxed.tsx:59]: sprinkles

File: src/boxed.tsx [hunk 5]
```diff
@@ -54,16 +40,25 @@ export const Boxed = React.forwardRef<HTMLDivElement, Props>(
...
+                    border: getBorderStyle(isInverseOutside, isInverseInside),
```
Comment (atabel) [inline src/boxed.tsx:60]: create classes for the border cases

**Review by pladaria — APPROVED** [review]

**Review by atabel — APPROVED** [review]

**Review by atabel — APPROVED** [review]

**Pull request merged**